java/ql/src/Security/CWE/CWE-926/ContentProviderIncompletePermissions.qhelp

```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>In the Android manifest file, an application's <code>provider</code> elements
+  define the permissions necessary to access a resource using that provider.
+  Permissions are specified with
+  the <code>android:readPermission</code>, <code>android:writePermission</code>,
+  or <code>android:permission</code> attributes. If an application only
+  specifies the <code>android:readPermission</code>
+  or <code>android:writePermission</code> attribute, no permissions will be
+  required to do other operations.
+</p>
+
+</overview>
+
+<recommendation>
+  <p>To prevent permission bypass, <code>provider</code> elements should either
+  specify both the <code>android:readPermission</code>
+  and <code>android:writePermission</code> attributes, or specify
+  the <code>android:permission</code> attribute.
```

Comment:
Maybe:

```suggestion
  <p>To prevent permission bypass, you should create <code>provider</code> elements that either
  specify both the <code>android:readPermission</code>
  and <code>android:writePermission</code> attributes, or specify
  the <code>android:permission</code> attribute.
```